common/src/firebase/common-actions.ts

```diff
@@ -0,0 +1,61 @@
+import { FirestoreCommon, FirestoreTag } from '../types/firestore-types';
+import { Tag } from '../types/store-types';
+import Database from './database';
+import OrderManager from './order-manager';
+import { ignore } from '../util/general-util';
+import { NONE_TAG_ID } from '../util/tag-util';
+
+type WithoutIdOrder<Props> = Pick<Props, Exclude<keyof Props, 'id' | 'order'>>;
+
+export default class Actions {
+  readonly orderManager: OrderManager;
+
+  constructor(private readonly getUserEmail: () => string, private readonly database: Database) {
```

Comment:
We need to delay `getUserEmail` evaluation since it is only available after auth.